### ISSUE TITLE
doesn't delete the oldest executions

### ISSUE THREAD

**Issue by firstway** [victorcoder/dkron#328]
https://github.com/victorcoder/dkron/blob/master/dkron/store.go#L359

I found that it  might delete the latest execution, which then causes that we miss the notification for that latest execution.

**Comment by firstway** [comment]
How to recreate the issue:
1. create a new test job,
```{
    "name": "job_name",
    "command": "/bin/true",
    "schedule": "@every 2m",
    "tags": {
        "role": "web:1"
    }
}
```
2, run the job many times until the number of execution groups exceeds `MaxExcutions` https://github.com/victorcoder/dkron/blob/master/dkron/store.go#L17 .
then the oldest execution groups are supposed to be deleted, so for the testing purpose, you can change this constant to some smaller value, (e.g. 10).   
With original code, not the oldest execution groups will be deleted, it's due to the fact the `execs` is unsorted.

**Comment by victorcoder** [comment]
@firstway Good catch!

**Comment by victorcoder** [comment]
Fixed in #331
